### DISCUSSION TITLE
What is the purpose of the `type` property of an action?

### DISCUSSION THREAD

**Discussion by scaljeri** [ngxs/store#1743] [Q&A] [answered]
### Description

Each action has a static property named `type`, but reading the docs I get the impression it is never used or described what its purpose is. I asked this question first on [SO](https://stackoverflow.com/questions/65658875/why-is-there-a-property-type-on-a-ngxs-action) and it was mentioned that it probably is only used internally to be able to connect the action with a state function. Although this might explain why I don't see it used anywhere, I think that some additional explanation should be in the docs why this property exists in the first place. Also, if this is true, and that property is used for identifying, why not use the action class as an identifier instead and remove that property?

**Comment by arturovt** [comment, marked as answer]
This is actually a classic Redux:

> An action is a plain JavaScript object that has a type field. You can think of an action as an event that describes something that happened in the application.

Not sure it's worth copy-pasting the already existing description.

More info here: https://redux.js.org/tutorials/fundamentals/part-2-concepts-data-flow

NgRx and other Redux-like state management decisions also require defining `type` property on actions.

**Comment by markwhitfeld** [comment]
@scaljeri regarding your question:
> why not use the action class as an identifier instead and remove that property?

This is a great idea, one we actually explored in the very early days of the library. Unfortunately this does not work after the minifier has done it's thing to the code for production builds. I can't remember the exact issue, because I was not involved in the investigation. But, if we could find a way to do this it would be really awesome! Ideas & PRs welcome 😉

**Reply by markwhitfeld** [reply]
I think that the major issue to solve is potentially around lazy-loaded states and actions that are referenced within/outside of those lazy states. Just thinking out loud here.

**Reply by scaljeri** [reply]
I must add that I'm not really a ngxs guru :) So I guess that this `type` solves the issues you mention, but now I'm curious what exactly the issue is it solves. @markwhitfeld can you gives some more details on that lazy-load if you have any. What exactly is the setup here, so I can correctly set this up!
